qiskit/circuit/library/grover_operator.py

```diff
@@ -0,0 +1,235 @@
+        state_3: ┤ Z ├┤ RY(-0.5) ├┤ RY(-0.1) ├┤ X ├┤ H ├┤ X ├┤ H ├┤ X ├┤ RY(0.1) ├┤ RY(0.5) ├
+                 └───┘└──────────┘└──────────┘└───┘└───┘└───┘└───┘└───┘└─────────┘└─────────┘
+
+        >>> mark_state = Statevector.from_label('011')
+        >>> diffuse_operator = 2 * DensityMatrix.from_label('000') - Operator.from_label('III')
+        >>> grover_op = GroverOperator(oracle=mark_state, zero_reflection=diffuse_operator)
+        >>> grover_op.draw(fold=70)
+                 ┌─────────────────┐      ┌───┐                          »
+        state_0: ┤0                ├──────┤ H ├──────────────────────────»
+                 │                 │┌─────┴───┴─────┐     ┌───┐          »
+        state_1: ┤1 UCRZ(0,pi,0,0) ├┤0              ├─────┤ H ├──────────»
+                 │                 ││  UCRZ(pi/2,0) │┌────┴───┴────┐┌───┐»
+        state_2: ┤2                ├┤1              ├┤ UCRZ(-pi/4) ├┤ H ├»
+                 └─────────────────┘└───────────────┘└─────────────┘└───┘»
+        «         ┌─────────────────┐      ┌───┐
+        «state_0: ┤0                ├──────┤ H ├─────────────────────────
+        «         │                 │┌─────┴───┴─────┐    ┌───┐
+        «state_1: ┤1 UCRZ(pi,0,0,0) ├┤0              ├────┤ H ├──────────
+        «         │                 ││  UCRZ(pi/2,0) │┌───┴───┴────┐┌───┐
+        «state_2: ┤2                ├┤1              ├┤ UCRZ(pi/4) ├┤ H ├
+        «         └─────────────────┘└───────────────┘└────────────┘└───┘
+
+    References:
+        [1]: L. K. Grover (1996), A fast quantum mechanical algorithm for database search,
+            `arXiv:quant-ph/9605043 <https://arxiv.org/abs/quant-ph/9605043>`_.
+        [2]: I. Chuang & M. Nielsen, Quantum Computation and Quantum Information,
+            Cambridge: Cambridge University Press, 2000. Chapter 6.1.2.
+        [3]: Brassard, G., Hoyer, P., Mosca, M., & Tapp, A. (2000).
+            Quantum Amplitude Amplification and Estimation.
+            `arXiv:quant-ph/0005055 <http://arxiv.org/abs/quant-ph/0005055>`_.
+    """
+
+    def __init__(self, oracle: Union[QuantumCircuit, Statevector],
+                 state_in: Optional[QuantumCircuit] = None,
+                 zero_reflection: Optional[Union[QuantumCircuit, DensityMatrix, Operator]] = None,
+                 reflection_qubits: Optional[List[int]] = None,
+                 insert_barriers: bool = False,
+                 mcx_mode: str = 'noancilla',
+                 name: str = 'Q') -> None:
+        """
+        Args:
+            oracle: The oracle implementing a reflection about the bad state.
+            state_in: The operator preparing the good and bad state. For Grover's algorithm,
+                this is a n-qubit Hadamard gate and for Amplitude Amplification or Estimation
+                the operator A.
+            zero_reflection: The reflection about the zero state.
+            reflection_qubits: Qubits on which the the zero reflection act on.
```

Comment:
There are a couple of minor grammar errors here (word repetition / act -> acts).

Reply:
resolved in 6464bc4